packages/devtools-reps/src/reps/tests/long-string.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { shallow } = require("enzyme");
+const {
+  REPS,
+  getRep,
+} = require("../rep");
+let { LongStringRep } = REPS;
+const stubs = require("../stubs/long-string");
+
+function quoteNewlines(text) {
+  return text.split("\n").join("\\n");
+}
+
+describe("test Multiline / LongStringRep", () => {
+  it("selects LongString Rep", () => {
+    const stub = stubs.get("testMultiline");
+    expect(getRep(stub)).toEqual(LongStringRep.rep);
+  });
+
+  it("renders with expected text content for multiline string", () => {
+    const stub = stubs.get("testMultiline");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub
+    }));
+    expect(renderedComponent.text()).toEqual(quoteNewlines(`"${stub.initial}…"`));
+  });
+
+  it("renders with expected text content for multiline string with " +
+    "specified number of characters", () => {
+    const stub = stubs.get("testMultiline");
+
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(`"a\\naaaaaaaaaaaaaaaaaa…"`);
+  });
+
+  it("renders with expected text content for multiline string when open", () => {
+    const stub = stubs.get("testMultiline");
+
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      member: {open: true},
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text())
+      .toEqual(quoteNewlines(`"${stub.initial}…"`));
+  });
+
+  it("renders with expected text content when grip has a fullText" +
+    "property and is open", () => {
+    const stub = stubs.get("testFullText");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      member: {open: true},
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(quoteNewlines(`"${stub.fullText}"`));
+  });
+
+  it("renders with expected text content when grip has a fullText " +
+    "property and is not open", () => {
+    const stub = stubs.get("testFullText");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(`"a\\naaaaaaaaaaaaaaaaaa…"`);
+  });
+
+  it("expected to omit quotes", () => {
+    const stub = stubs.get("testMultiline");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20,
+      useQuotes: false
+    }));
+
+    expect(renderedComponent.text()).toEqual(`a aaaaaaaaaaaaaaaaaa…`);
```

Comment:
@nchevobbe - I started refactoring with that original assertion `"a\naaaaaaaaaaaaaaaaaa…"` but that results with:
``` FAIL  src/reps/tests/long-string.js
  ● LongStringRep › expected to omit quotes

    expect(received).toEqual(expected)
    
    Expected value to equal:
      "a
    aaaaaaaaaaaaaaaaaa…"
    Received:
      "a aaaaaaaaaaaaaaaaaa…"
      
      at Object.it (src/reps/tests/long-string.js:87:38)
      at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)
```
therefore I started thinking if the original assertion was correct. The test result indicates that the function returns one line, but the assertion states that the result should contain a line break. That's why I wasn't sure what's wrong, the function or the assertion? 
Another possibility is that I called it incorrectly - but I hope that's not the case.